src/positions/selectors.ts

```diff
@@ -37,3 +42,43 @@ function sortByBalanceUsd(position1: Position, position2: Position) {
 export const positionsByBalanceUsdSelector = createSelector([positionsSelector], (positions) => {
   return [...positions].sort(sortByBalanceUsd)
 })
+
+export const shortcutsSelector = (state: RootState) => state.positions.shortcuts
+
+export const claimableShortcutSelector = createSelector([shortcutsSelector], (shortcuts) => {
+  return shortcuts.filter((shortcut) => shortcut.category === 'claim')
+})
+
+function getAllClaimableTokens(tokens: TokenWithCategory[]): TokenWithCategory[] {
+  const claimableTokens = tokens.filter((token) => token.category === 'claimable')
+  const nestedTokens = tokens
+    .filter((token): token is AppTokenPosition => 'tokens' in token)
+    .flatMap((token) => getAllClaimableTokens(token.tokens))
+
+  return [...claimableTokens, ...nestedTokens]
+}
+
+export const positionsWithClaimableRewardsSelector = createSelector(
+  [positionsSelector, claimableShortcutSelector],
+  (positions, shortcuts) => {
+    const claimablePositions: ClaimablePosition[] = []
+    positions.forEach((position) => {
+      const appShortcutIds = shortcuts
+        .filter((shortcut) => shortcut.appId === position.appId)
```

Comment:
For now we only want `claim` shortcuts.
In the future it's possible we'll have a position with a claimable token and a claim shortcut but also other shortcuts (deposit, withdraw, etc).


```suggestion
        .filter((shortcut) => shortcut.appId === position.appId && shortcut.category === 'claim')
```